goclean.sh

```diff
@@ -8,6 +8,32 @@
 
 set -ex
 
+# Check if we have gcc, if not then use clang, else exit. More compilers can be
+# added if needed.
+if command -v gcc &> /dev/null
+then
+
+    export CC=gcc
+
+elif command -v clang &> /dev/null
+then
+
+    export CC=clang
```

Comment:
This is fine, but we should still allow the user to override the C compiler using the `CC` env var. What do you think?

Reply:
That makes sense, I'll change it so that if the user has the `CC` env variable set it will use that C compiler.

Reply:
Not sure if there could be situations when `go env` has an empty `CC` variable, it is set to `gcc` by default even if `gcc` is not installed.
Also, current behavior gonna overwrite `CC` variable to be `gcc` for me as I'm using `CC=clang` compiler while having `gcc` and `clang` both installed.

Reply:
You're right, I should change it so that it doesn't overwrite the CC variable after the script is run. However, there are situations (such as on my current machine) where the `CC` environment variable in `go env` is set to `gcc` when `gcc` is not installed, leading to tests crashing which is the point of the PR.

Reply:
Yeah, I'm understanding your idea why you've made this PR and that's why I'm asking you to be more careful with such changes as they can break things for other users. Just make sure it's gonna be error-safe after you make it right, thanks :)